cross-account-region-data-pipeline/README.md

```diff
@@ -0,0 +1,391 @@
+
+There are three environments to bootstrap for each of their respective regions and accounts (`cicd`, `stor`, `comp`). To get an understanding of the parameter options for the bootstrap command, first run:
+```
+$ ddk bootstrap --help
+```
+
+The `cicd` environemnt is where the source repository and CICD Pipeline will exist for your application. Run the following command to bootstrap:
+<br />
+```
+$ ddk bootstrap --profile [AWS_PROFILE] --region [AWS_REGION] -e cicd --trusted-accounts [OTHER_AWS_ACCOUNT]
+```
+
+The `stor` environment is where the storage S3 Bucket and related resources (i.e. KMS Key, etc.) will exist for your application. Run the following command to bootstrap:
+<br />
+```
+$ ddk bootstrap --profile [AWS_PROFILE] --region [AWS_REGION] -e stor --trusted-accounts [OTHER_AWS_ACCOUNT]
+```
+
+The `comp` environemnt is where the compute resources (i.e. SQS, Lambda, etc.) will exist for your application. Run the following command to bootstrap:
+<br />
+```
+$ ddk bootstrap --profile [AWS_PROFILE] --region [AWS_REGION] -e comp --trusted-accounts [OTHER_AWS_ACCOUNT]
+```
+
+_NOTE: The `--trusted-accounts` parameters is only required for Cross Account patterns and can be called multiple times for passing multiple account IDs_
+
+### Editing ddk.json 
+
+Open the `ddk.json` file in the codebase and edit each of your `cicd`, `stor`, and `comp` environments with the correct `account` and `region`
+
+- This Codebase supports ALL same account, cross account and cross region patterns for creating your data pipeline
+
+Edit the `s3BucketName` in both the `stor` and `comp` environments for the name of the S3 Bucket to create
+
+- The S3 Bucket Name should be the same for both `stor` and `comp` environments and MUST be globally unique to create a new S3 Bucket resource
+
+### Performing Git Operations 
+<br />
+
+Initialise git for the repository
+
+```
+$ git init --initial-branch main
+```
+
+Execute the create repository command to create a new codecommit repository in the aws account for the `cicd` environment
+
+_NOTE: Make Sure the REPO_NAME matches the repository name value in the app.py file or vice versa before executing_
+
+```
+$ ddk create-repository REPO_NAME --profile [AWS_PROFILE] --region [AWS_REGION]
+```
+
+Add and push the initial commit to the repository
+
+```
+$ git add .
+$ git commit -m "Configure ADK"
+$ git push --set-upstream origin main
+```
+
+### Deploying the Data Processing Pipeline
+
+Verify the `ddk.json` is updated with your configuration and run the deploy command to deploy:
+
+```
+$ ddk deploy --profile [AWS_PROFILE]
+```
+
+The AWS_PROFILE is the profile for the `cicd` environment to deploy in the same account and region as the source CodeCommit Repository
+
+#
+
+## Cross Account Cross Region Pipeline
```

Comment:
It's a bit hard to tell that there's three different examples here. Can we change header size or add like a table of contents of each example after the initial setup section?